o11y/tracer/datadog/datadog.go

```diff
@@ -0,0 +1,80 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2023 Datadog, Inc.
+
+package datadog
+
+import (
+	"strconv"
+
+	"github.com/DataDog/chaos-controller/o11y/tracer/types"
+	"go.opentelemetry.io/otel/trace"
+
+	ddotel "gopkg.in/DataDog/dd-trace-go.v1/ddtrace/opentelemetry"
+	ddtracer "gopkg.in/DataDog/dd-trace-go.v1/ddtrace/tracer"
+)
+
+// Sink describes a datadog tracer sink
+type Sink struct {
+	provider *ddotel.TracerProvider
+}
+
+// New initiated datadog tracer sink
+func New() Sink {
+	provider := ddotel.NewTracerProvider(
+		ddtracer.WithProfilerCodeHotspots(true),
+		ddtracer.WithLogStartup(false),
+	)
+
+	return Sink{provider: provider}
+}
+
+// GetLoggableTraceContext returns a Datadog-friendly trace context for logs
+// It allows to connect logs with corresponding traces and spans
+func (Sink) GetLoggableTraceContext(span trace.Span) []interface{} {
+	stringLogContext := []string{
+		"dd.trace_id", convertTraceID(span.SpanContext().TraceID().String()),
+		"dd.span_id", convertTraceID(span.SpanContext().SpanID().String()),
+	}
+
+	interfaceLogContext := make([]interface{}, len(stringLogContext))
+	for i, v := range stringLogContext {
+		interfaceLogContext[i] = v
+	}
+
+	return interfaceLogContext
+}
+
+// GetProvider returns this sink's provider for otel
+func (d Sink) GetProvider() trace.TracerProvider {
+	return d.provider
+}
+
+// Stop shutdowns the datadog provider for otel
+func (d Sink) Stop() error {
+	return d.provider.Shutdown()
+}
+
+// GetSinkName returns the name of the sink (datadog)
+func (d Sink) GetSinkName() string {
+	return string(types.SinkDriverDatadog)
+}
+
+// convertTraceID converts a OTLP Trace/Span ID into a DataDog Trace/Span ID
+func convertTraceID(id string) string {
+	if len(id) < 16 {
```

Comment:
🤔 And Datadog SDK does not provide anything to do such things for us?

Reply:
according to [this doc](https://docs.datadoghq.com/tracing/other_telemetry/connect_logs_and_traces/opentelemetry/?tab=go) it doesn't seem so 🤷🏻‍♂️ 
after investigation, I didn't find anything relevant in the ddtrace library :/